kernel/work_q.c

```diff
@@ -125,6 +125,12 @@ int k_delayed_work_cancel(struct k_delayed_work *work)
 	}
 
 	if (k_work_pending(&work->work)) {
+                /*
+                 * [Sanechips] clear bit of work.flags,
+                 * when remove from the queue if already submitted.
+                 */
+		atomic_clear_bit(work->work.flags,K_WORK_STATE_PENDING);
```

Comment:
You don't need to use an atomic operation here, interrupts are locked anyway.

Reply:
Also why are we doing this here now? Previously, work_q_main() would clear this bit when it actually starts doing the work? I do not understand why this is also done here now.